### ISSUE TITLE
Decorative Images

### ISSUE THREAD

**Issue by ways2read** [ways2read/InDesignA11Y#26]
Knowledge base reference: http://kb.daisy.org/publishing/docs/html/images-deco.html

**Comment by ways2read** [comment]
User story: "As a screen reader user I am not bothered by hearing that images are in the publication that do not have image descriptions, and wondering if it is important content that has not been described."

**Comment by raman211** [comment]
Hi @gregoriopellegrino 
This is Ramanjulu from Adobe InDesign Team. I am looking to fix the above issue and need some clarity from you.

The HTML snippet for the Decorative image suggested, as per the above kb article i.e
```
<img
    src="graphics/gothic-border.png"
    role="presentation"
    alt=""/> 
```
 doesn't seem to be a valid HTML code snipper according to the https://validator.w3.org/
Should This be just specific to EPUB? For HTML should there only be an empty Alt Tag without any ARIA role??
Could you please let us know, what is expected here for the Decorative Images??

**Comment by ways2read** [comment]
This is a known issue with the HTML validator. It was [reported last year](https://github.com/validator/validator/issues/1425). There's an interesting thread there.

However, this is only an issue in the HTML validator, so doesn’t affect EPUB. Epubcheck doesn’t report role=presentation on images with empty alt attributes.

So, it’s not invalid to add the role to InDesign’s output for HTML and EPUB.

**Comment by gregoriopellegrino** [comment]
We have seen the implementation in InDesign 18.5: images without alternate text are exported as decorative images (`role = "presentation"`). We thank you for the development, at the same time we are concerned that an oversight by the layout designer may lead to the lack of alternative text, thus making an image that should not be decorative.

Indeed, it is one thing to designate an image as decorative, and quite another to accidentally forget to include the alternative text. To avoid potential errors, we suggest implementing a more explicit control (a checkbox?), similar to the option available for PDFs where `Artifact` tags can be assigned, so that decorative images can be clearly identified.

**Comment by Racmathu** [comment]
@gregoriopellegrino I am Rachit from Adobe. We understand your concern and we are working with our User Experience team to check how we could provide such controls to you. We will get back to you soon on this.

**Comment by raman211** [comment]
@gregoriopellegrino @ways2read @LauraB7 Could you please look at the following proposed UX and share your thoughts on it

<img width="565" alt="DecorativeImage2" src="https://github.com/ways2read/InDesignA11Y/assets/12660080/5dd4ba07-b5d7-415b-90a6-69174d361ca2">

<img width="593" alt="DecorativeImage1" src="https://github.com/ways2read/InDesignA11Y/assets/12660080/4e0f5e33-0302-4b2a-8563-85c948147670">

Thanks,
Ramanjulu.

**Comment by gregoriopellegrino** [comment]
Thanks @raman211 it is fine for use. We'll wait for testing it soon.

**Comment by raman211** [comment]
Hi @gregoriopellegrino @ways2read @LauraB7,

The feature is available in the latest InDesign PreRelease of 19.4
Looking forward to your valuable feedback, before releasing it in 19.4
Please share your feedback on it ASAP, so that we can accommodate the changes for the 19.4 release.

Thanks,
Ramanjulu.

**Comment by gregoriopellegrino** [comment]
@raman211 it works well!

A further improvement can be that the option “Custom”, which is the first alternative in the alt text dropdown, may be the default, pre-selected alternative (now it is “From structure”).

In any case, the issue is fine for us, we wait for the official release to close it.

**Comment by Racmathu** [comment]
Thanks Greg. Please file a JIRA ticket in case you find a bug in our implementation OR you need some enhancements/improvements. That way, you'll have visibility into the latest updates on that issue.

**Comment by raman211** [comment]
Hi @gregoriopellegrino @ways2read @LauraB7 @awkawk @jonaslil 

Please find the changes in the ID19.4 release builds, which are available for the public now.
CC: @Racmathu

**Comment by gregoriopellegrino** [comment]
The functionality works well, but we still ask a further improvement: the option “Custom”, which is the first alternative in the alt text dropdown, may be the default, pre-selected alternative (now it is “From structure”).
